src/display/annotation_layer.js

```diff
@@ -758,28 +792,37 @@ class PopupAnnotationElement extends AnnotationElement {
     }
 
     const selector = `[data-annotation-id="${this.data.parentId}"]`;
-    const parentElement = this.layer.querySelector(selector);
-    if (!parentElement) {
+    const parentElement = this.layer.querySelectorAll(selector);
+    if (!parentElement || parentElement.length === 0) {
       return this.container;
     }
 
     const popup = new PopupElement({
       container: this.container,
-      trigger: parentElement,
+      trigger: Array.from(parentElement),
       color: this.data.color,
       title: this.data.title,
       modificationDate: this.data.modificationDate,
       contents: this.data.contents,
     });
 
+    // Find the main container if any
+    let main = parentElement.item(0);
```

Comment:
I don't see why we need this and the `first` logic above. The main container should always be `this.container` and you can access it everywhere. Moreover, we should not be storing a non-default property like `mainContainer` on a standard element.

Reply:
Popup was positioned according the highlight container but now the highlights are splited. So my goal was to position the Popup using the parent rect finally.
So instead of doing that, I added parent rect in `Popup.data` (in `annotation.js`).